tfx_bsl/beam/run_inference.py

```diff
@@ -507,12 +701,26 @@ class _BaseBatchSavedModelDoFn(_BaseDoFn):
 
   def __init__(
       self,
-      inference_spec_type: model_spec_pb2.InferenceSpecType,
       shared_model_handle: shared.Shared,
+      fixed_inference_spec_type: model_spec_pb2.InferenceSpecType = None
   ):
-    super(_BaseBatchSavedModelDoFn, self).__init__(inference_spec_type)
-    self._inference_spec_type = inference_spec_type
+    super(_BaseBatchSavedModelDoFn, self).__init__(fixed_inference_spec_type)
     self._shared_model_handle = shared_model_handle
+    self._fixed_inference_spec_type = fixed_inference_spec_type
+    self._use_fixed_model = (self._fixed_inference_spec_type is not None)
+
+  def setup(self):
+    """Load the model.
+    Note that worker may crash if exception is thrown in setup due
+    to b/139207285.
+    """
+    super(_BaseBatchSavedModelDoFn, self).setup()
+    if self._use_fixed_model:
+      self._setup_model(self._fixed_inference_spec_type)
+
+  def _setup_model(
+      self, inference_spec_type: model_spec_pb2.InferenceSpecType
+  ):
     self._model_path = inference_spec_type.saved_model_spec.model_path
     self._tags = None
```

Comment:
Move the initialization of _tags, _session, _io_tensor_spec to __init__. And initialize _model_path, _signatures to None in __init__.